satellite/gracefulexit/endpoint.go

```diff
@@ -335,17 +335,17 @@ func (endpoint *Endpoint) processIncomplete(ctx context.Context, stream processS
 	}
 
 	if len(newNodes) == 0 {
-		return Error.New("could not find a node to transfer this piece to. nodeID %v, path %v, pieceNum %v.", nodeID.String(), zap.ByteString("path", incomplete.Path), incomplete.PieceNum)
+		return Error.New("could not find a node to receive piece transfer: node ID %v, path %v, piece Num %v", zap.Stringer("node ID", nodeID), zap.ByteString("path", incomplete.Path), zap.Int32("piece num", incomplete.PieceNum))
 	}
 	newNode := newNodes[0]
-	endpoint.log.Debug("found new node for piece transfer.", zap.String("original node ID", nodeID.String()), zap.String("replacement node ID", newNode.Id.String()),
+	endpoint.log.Debug("found new node for piece transfer", zap.Stringer("original node ID", nodeID), zap.Stringer("replacement node ID", newNode.Id),
 		zap.ByteString("path", incomplete.Path), zap.Int32("piece num", incomplete.PieceNum))
 
 	pieceID := remote.RootPieceId.Derive(nodeID, incomplete.PieceNum)
 
 	parts := storj.SplitPath(storj.Path(incomplete.Path))
 	if len(parts) < 2 {
-		return Error.New("invalid path for %v %v.", zap.String("node ID", incomplete.NodeID.String()), zap.String("pieceID", pieceID.String()))
+		return Error.New("invalid path for node ID %v, piece ID %v", zap.Stringer("node ID", incomplete.NodeID), zap.Stringer("piece ID", pieceID))
```

Comment:
Only logging should use zap values. This should be:

```
Error.New("invalid path for node ID %v, piece ID %v", incomplete.NodeID, pieceID)
```

There probably are other places that have this mistake.

Reply:
oh oops, I'll fix. thanks for catching.

Reply:
@egonelbre should the errors use nodeID.String() or just nodeID?

Reply:
It's the same thing. If you use `%v` or `%s` then Go will first try to use `.String()` method.